lib/palindrome_check.rb

```diff
@@ -1,7 +1,34 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n) where n is length of my_phrase
+# Space complexity: 0(1)
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+
+  if my_phrase.length == 0
+    return true
+  end
+
+  i = 0
+  j = my_phrase.length - 1
+
+  while i < j
```

Comment:
This `while` loop is nicely structured. Keep in mind how you implemented this; I've seen similar sorts of structures in interview questions before.